asmcomp/riscv/proc.ml

```diff
@@ -241,25 +240,27 @@ let regs_are_volatile _ = false
 
 (* Registers destroyed by operations *)
 
-let destroyed_at_c_call =
-  (* s0-s11 and fs0-fs11 are callee-save.  However s2 needs to be in this
-     list since it is clobbered by caml_c_call itself. *)
+let destroyed_at_c_noalloc_call =
+  (* s0-s11 and fs0-fs11 are callee-save, but s0 is
+     used to preserve OCaml sp. *)
   Array.of_list(List.map phys_reg
-    [0; 1; 2; 3; 4; 5; 6; 7; 8; 16; 17; 18; 19; 20; 22;
+    [0; 1; 2; 3; 4; 5; 6; 7; 16; 17; 18; 19; 20; 21 (* s0 *);
```

Comment:
For posterity, could you tell us why `8` and `22` are gone and `21` added? A reply to this question here is fine. No need to add comments in the code.

Reply:
- `8` (= `s2`) is removed: the comment in `trunk` indicates that this register was included in this list because it is clobbered by `caml_c_call`. However this seems like a mistake, because this list specifies those registers killed by `noalloc` calls (basically caller-save ones), which do not go through `caml_c_call` at all. So it looks like `8` did not need to be here at all to begin with.
- `21` (= `s0`) is added: this is explained in the discussion [above](https://github.com/ocaml/ocaml/pull/11418#discussion_r1012663868)
- `22` (= `t0`) is removed: this one is now [used](https://github.com/nojb/ocaml/blob/71b9ebae167c5574b25d18375df5c45715a0d787/runtime/riscv.S#L27) in the runtime stubs as a general auxiliary register, so it is removed from the set of allocatable registers altogether (realized by [lowering](https://github.com/nojb/ocaml/blob/riscv_5/asmcomp/riscv/proc.ml#L89) `num_available_registers.(0)` from 23 to 22).